### PR TITLE
Add emoji support for repository description

### PR DESCRIPTION
**Pull request by kunyan**
#254  because search result is render by mustache, we can not use [jemoji](https://github.com/jekyll/jemoji)

http://www.emoji-cheat-sheet.com/  Emoji sheet

/\>.*\:([a-z_+-]*)\:/   https://regexper.com/#%2F%5C%3E.*%5C%3A(%5Ba-z_%2B-%5D*)%5C%3A%2F

Only replace content

### CODE REVIEW

**Comment by julmot** [comment]
Emojis might be worth considiering as a feature, but I'd prefer to discuss this first in #254. 

However, I personally see a few issues with your changes:
- You're using `innerHTML` a second time, which triggers re-generation of the DOM also a second time. It could be avoided by applying the replacement to either the result of the rendered Mustache code or directly to the (cloned) state
- The regex seems to not work correctly with multiple smileys (see e.g. https://regex101.com/r/hD3qR0/1), it will also select everything between two smileys
- You're mixing HTML with JS. To avoid confusion and to reduce this huge line it might be better to put this into a separate Mustache template

**Comment by kunyan** [comment]
@julmot  thanks for review my code :smile:  .  I fix all issues you mentioned.

File: search/index.html [hunk 1]
```diff
@@ -53,4 +53,7 @@
     {{/flash}}
   </div>
 </script>
+<script id="emoji-template" type="x-tmpl-mustache">
+<img class="emoji" title=":{{code}}:" alt=":{{code}}:" src="https://assets-cdn.github.com/images/icons/emoji/{{code}}.png" height="20" width="20" align="absmiddle"/>
```
Comment (julmot) [inline search/index.html]: I've misread this on first commit, thought it was a class.

> The align attribute of  is not supported in HTML5

See e.g. http://www.w3schools.com/tags/att_img_align.asp

File: js/index.js [hunk 2]
```diff
@@ -120,6 +120,15 @@ function renderStats() {
   });
 }
 
+function renderEmoji(code) {
+  var template = document.getElementById('emoji-template').innerHTML;
```
Comment (julmot) [inline js/index.js]: Getting the element over and over again might be an overhead. Just cache it like the others [here](https://github.com/kunyan/bower.github.io/blob/fa02f809c894269bb63abc1a2d2cca492fcfebd8/js/index.js#L145).

File: js/index.js [hunk 3]
```diff
@@ -120,6 +120,15 @@ function renderStats() {
...
+  return Mustache.render(template, {code: code});
+}
+
+function filterEmoji(str) {
+  return str.replace(/\:([a-z_+\-0-9]*)\:/g, renderEmoji('$1'));
```
Comment (julmot) [inline js/index.js]: There are still issues with this, e.g.
https://regex101.com/r/bR4eY2/1

You need to exclude matches within HTML tags (e.g. positive lookahead and non-capturing group).

**Comment by julmot** [comment]
I've added some new comments. However, before spending more time on this we should discuss if we want such feature first (as mentioned in my first comment).

**Comment by ZDroid** [comment]
:+1:

File: js/index.js [hunk 4]
```diff
@@ -171,6 +172,14 @@ function renderSearch() {
   }
   search = _.debounce(search, 1000);
 
+  function renderEmoji(code) {
+    return Mustache.render(emojiTemplate, {code: code});
+  }
+
+  function filterEmoji(str) {
+    return str.replace(/\:([a-z_+\-0-9]*)\:/g, renderEmoji('$1'));
```
Comment (kunyan) [inline js/index.js]: @julmot  I didn't find a best way to filter html tag context only. we should not insert any emoji symbol to html tag attribute.

**Comment by julmot** [comment]
@kunyan Thank you very much for your contribution. I had to format the code and fix the line break of emojis. However, now everything seems to be fine! :beers:

**Pull request closed without merging**